### PR TITLE
NAD-973 Stack component

### PR DESCRIPTION
**Pull request by matheusps**
#### What is the purpose of this pull request?

- Add stack component
- Replace usages of Set inside the admin-ui package for Stack or Inline.

The Set component was not removed right now to not block our users. Added a new issue to track the Set deprecation: https://vtex-dev.atlassian.net/jira/software/c/projects/NAD/boards/300/backlog?view=detail&selectedIssue=NAD-1098&epics=visible&issueLimit=100&selectedEpic=NAD-857. 

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

**Review by anitavincent — APPROVED** [review]

File: packages/admin-ui-docs/docs/guidelines/components/stack.mdx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    mobile: 'column',
+    tablet: 'row',
+  }}
+  fluid
+>
+  <Button>First</Button>
+  <Button>Second</Button>
+  <Button>Third</Button>
+  <Button>Fourth</Button>
+  <Button>Fith</Button>
```
Comment (Lenderson) [inline packages/admin-ui-docs/docs/guidelines/components/stack.mdx:94]: typo here: **Fifth**

File: packages/admin-ui-docs/docs/guidelines/components/stack.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+</Stack>
+```
+
+</Panel>
+
+<Panel id="codeReference">
+
+## Usage
+
+```jsx isStatic
```
Comment (Lenderson) [inline packages/admin-ui-docs/docs/guidelines/components/stack.mdx:104]: No need to use isStatic, it is the default mode 😉

File: packages/admin-ui/src/inline/inline.tsx [hunk 3]
```diff
@@ -4,15 +4,23 @@ import type {
...
+    align = 'start',
+    ...htmlProps
+  } = props
 
   return useElement('div', {
     ...htmlProps,
     baseStyle: {
       display: 'flex',
-      flexWrap: 'wrap',
+      flexWrap: noWrap ? 'nowrap' : 'wrap',
```
Comment (Lenderson) [inline packages/admin-ui/src/inline/inline.tsx:22]: **Suggestion**: I know `nowrap` is a `flex-wrap` valid value, but trying to give a more explicit to the dev about this prop type, maybe we could change its name to something like: `isNoWrap` or `isWrapped` 🤔

File: packages/admin-ui/src/stack/responsive.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+/**
+ * React hook that tracks state of a CSS media query
+ */
+export function useBreakpoint() {
+  const isMatchMediaSupported = isBrowser && 'matchMedia' in window
+
+  const [matches, setMatches] = useState(
+    queries.map((query) =>
+      isMatchMediaSupported ? window.matchMedia(query).matches : false
```
Comment (Lenderson) [inline packages/admin-ui/src/stack/responsive.ts:16]: Suggestion:
```suggestion
      isMatchMediaSupported && window.matchMedia(query).matches
```

File: packages/admin-ui/src/stack/stack.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  /**
+   * direction of items
+   * @default column
+   */
+  direction?: ResponsiveProp<'column' | 'row'>
+  /**
+   * if the items should grow in width to match the container
+   * @default false
+   */
+  fluid?: ResponsiveProp<boolean>
```
Comment (Lenderson) [inline packages/admin-ui/src/stack/stack.ts:77]: **Suggestion: ** Similar to[ this point](https://github.com/vtex/admin-ui/pull/135/files#r852316814) could it make sense to change to something like: `isFluid`

**Review by lucasaarcoverde — APPROVED** [review]

File: packages/admin-ui/src/inline/inline.tsx (hunk 3 above)
Comment (matheusps) [reply packages/admin-ui/src/inline/inline.tsx:22]: I will keep this pattern until we discuss it.

File: packages/admin-ui/src/stack/stack.ts (hunk 5 above)
Comment (matheusps) [reply packages/admin-ui/src/stack/stack.ts:77]: Same

**Pull request merged**